core/src/stored/ndmp_tape.cc

```diff
@@ -387,51 +375,60 @@ static inline bool bndmp_read_data_from_block(JobControlRecord* jcr,
       rctx->lastFileIndex = READ_NO_FILEINDEX;
 
       if (!ReadNextRecordFromBlock(dcr, rctx, &done)) {
-        /*
-         * When the done flag is set to true we are done reading all
-         * records or end of block read next block.
-         */
+        // When the done flag is set to true we are done reading all
+        // records or end of block read next block.
         continue;
       }
     }
 
     // See if we are processing some sort of label?
     if (rctx->rec->FileIndex < 0) { continue; }
 
-    /*
-     * Here we should have read a record from the block which contains some
-     * data. Its either:
-     *
-     * - STREAM_UNIX_ATTRIBUTES
-     *      Which is the start of the dump when we encounter that we just read
-     * the next record.
-     * - STREAM_FILE_DATA
-     *      Normal NDMP data.
-     * - STREAM_NDMP_SEPARATOR
-     *      End of NDMP data stream.
-     *
-     * anything other means a corrupted stream of records and means we give an
-     * EOF.
-     */
-    switch (rctx->rec->maskedStream) {
-      case STREAM_UNIX_ATTRIBUTES:
+    DeviceRecord* rec = rctx->rec;
+    // Perform record translations only if data is compressed
+    // as NDMP needs to be decompressed in any case
+    if (rctx->rec->maskedStream == STREAM_COMPRESSED_DATA) {
+      dcr->before_rec = rctx->rec;
+      dcr->after_rec = NULL;
+
+      if (GeneratePluginEvent(jcr, bSdEventReadRecordTranslation, dcr,
+                              /* reverse= */ true)
+          != bRC_OK) {
+        ok = false;
+        continue;
+      }
+      if (dcr->after_rec) {  // translation happened?
+        rec = dcr->after_rec;
```

Comment:
if `GeneratePluginEvent()` set `dcr->after_rec`, doesn't that mean it allocated a new record that we need to free with `FreeRecord()`?